staging/src/k8s.io/kubectl/pkg/drain/cordon.go

```diff
@@ -72,7 +72,7 @@ func (c *CordonHelper) UpdateIfRequired(desired bool) bool {
 // updating the given node object; it may return error if the object cannot be encoded as
 // JSON, or if either patch or update calls fail; it will also return a second error
 // whenever creating a patch has failed
-func (c *CordonHelper) PatchOrReplace(clientset kubernetes.Interface, serverDryRun bool) (error, error) {
+func (c *CordonHelper) PatchOrReplace(clientCtx context.Context, clientset kubernetes.Interface, serverDryRun bool) (error, error) {
```

Comment:
I think it might be better to add a new method `PatchOrReplaceWithContext`, then invoke this method in `PatchOrReplace` with `context.TODO()`. 
And you could invoke this new method if you want to provide your own context.